<launch>

    <arg name="robot_name" default="sam"/>

    <node name="pid_actuator_tcg" pkg="sam_basic_controllers" type="pid_actuator_tcg" ns="tcg" >
      <param name="topic_from_controller_1" value="control_action" />
      <param name="topic_from_controller_2" value="control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="pid_enable_topic" value="/$(arg robot_name)/ctrl/tcg/pid_enable" />
      <param name="limit_between_setpoints" value="0.01" />
      <param name="loop_freq" value="50" />
      <remap from="/actuator_input" to="/$(arg robot_name)/core/tcg_cmd"/>
    </node>

    <node name="tcg_pid" pkg="pid" type="controller" ns="tcg" output="screen" >
      <param name="Kp" value="5.0" />
      <param name="Ki" value="5.0" />
      <param name="Kd" value="2.0" />
      <param name="upper_limit" value="3.1415" />
      <param name="lower_limit" value="-3.1415" />
      <param name="windup_limit" value="10" />
      <param name="max_loop_frequency" value="100.0" />
      <param name="min_loop_frequency" value="100.0" />
      <!-- CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! -->
      <param name="topic_from_plant" value="/$(arg robot_name)/dr/roll" />
      <!--param name="topic_from_plant" value="/$(arg robot_name)/ctrl/odom_listener/roll_feedback" /-->
      <param name="topic_from_controller" value="control_action" />
      <remap from="setpoint" to="setpoint" />
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
      <remap from="pid_enable" to="pid_enable" />
     </node>


</launch>
